vpr/src/route/rr_graph_generation/rr_graph2.cpp

```diff
@@ -2008,41 +1447,8 @@ int get_track_to_tracks(RRGraphBuilder& rr_graph_builder,
     return num_conn;
 }
 
-void alloc_and_load_tile_rr_node_indices(RRGraphBuilder& rr_graph_builder,
-                                         t_physical_tile_type_ptr physical_tile,
-                                         int layer,
-                                         int x,
-                                         int y,
-                                         int* num_rr_nodes) {
-    std::vector<e_side> wanted_sides{TOP, BOTTOM, LEFT, RIGHT};
-    auto class_num_range = get_flat_tile_primitive_classes(physical_tile);
-    auto pin_num_vec = get_flat_tile_pins(physical_tile);
-
-    std::vector<int> class_num_vec(class_num_range.total_num());
-    std::iota(class_num_vec.begin(), class_num_vec.end(), class_num_range.low);
-
-    add_classes_spatial_lookup(rr_graph_builder,
-                               physical_tile,
-                               class_num_vec,
-                               layer,
-                               x,
-                               y,
-                               physical_tile->width,
-                               physical_tile->height,
-                               num_rr_nodes);
-
-    add_pins_spatial_lookup(rr_graph_builder,
-                            physical_tile,
-                            pin_num_vec,
-                            layer,
-                            x,
-                            y,
-                            num_rr_nodes,
-                            wanted_sides);
-}
-
 static int get_bidir_track_to_chan_seg(RRGraphBuilder& rr_graph_builder,
-                                       const std::vector<int> conn_tracks,
+                                       const std::vector<int>& conn_tracks,
```

Comment:
Good catch .. shouldn't be passing vectors by value like this ... maybe we'll get some speed up from this.